docs/sdk/src/guides/enable_elastic_scaling.rs

```diff
@@ -0,0 +1,162 @@
+//!
+//! Depending on their use case, applications might have an increased need for the following:
+//! - compute (CPU weight)
+//! - bandwidth (proof size)
+//! - lower latency (block time)
+//!
+//! ### High throughput (TPS) and lower latency
+//!
+//! If the main bottleneck is the CPU, then your parachain needs to maximize the compute usage of
+//! each core while also achieving a lower latency.
+//! 3 cores provide the best balance between CPU, bandwidth and latency: up to 6s of execution,
+//! 3MB/s of DA bandwidth and fast block time of just 2 seconds.
+//!
+//! ### High bandwidth
+//!
+//! Useful for applications that are bottlenecked by bandwidth.
+//! By using 6 cores, applications can make use of up to 6s of compute, 6MB/s of bandwidth every 6s
+//! while also achieving 1 second block times.
+//!
+//! ### Ultra low latency
+//! When latency is the primary requirement, Elastic scaling is currently the only solution. The
+//! caveat is the efficiency of core time usage decreases as more cores are used.
+//!
+//! For example, using 12 cores enables fast transaction confirmations with 500ms blocks and up to
+//! 12 MB/s of DA bandwidth.
+//!
+//! ## Dependencies
+//!
+//! Prerequisites: Polkadot-SDK `2509` or newer.
+//!
+//! To ensure the security and reliability of your chain when using this feature you need the
+//! following:
+//! - An omni-node based collator. This has already become the default choice for collators.
+//! - [RFC103](https://github.com/polkadot-fellows/RFCs/blob/main/text/0103-introduce-core-index-commitment.md).
+//!   This is mandatory protection against PoV replay attacks.
+//! - Enabling the relay parent offset feature. This is required to ensure the parachain block times
+//!   and transaction in-block confidence are not negatively affected by relay chain forks.
+//! - Block production configuration adjustments.
+//!
+//! ### Upgrade to Polkadot Omni node
+//!
+//! Your collators need to run `polkadot-parachain` or `polkadot-omni-node` with the `--authoring
+//! slot-based` CLI argument.
+//!
+//! Further information about omni-node and how to upgrade is available:
+//! - [high level docs](https://docs.polkadot.com/develop/toolkit/parachains/polkadot-omni-node/)
+//! - [`crate::reference_docs::omni_node`]
+//!
+//! ### Enable RFC103
+//!
+//! RFC103 is enabled automatically on the collators if it is enabled on the relay chain. There are
+//! no code changes required on the client to support it.
+//!
+//! RFC103 relies on the ability of parachain blocks to commit to a specific core index on the relay
+//! chain. This commitment is implemented via `UMP` signals, which rely on the upward message
+//! passing channel that is used by parachains to send messages to the relay chain.
+//!
+//! The only required change for the runtime is enabling the `experimental-ump-signals` feature of
+//! the `parachain-system` pallet:
+//! `cumulus-pallet-parachain-system = { workspace = true, features = ["experimental-ump-signals"]
```

Comment:
when do we envision that we'll be able to remove this feature flag?
AFAICT we cannot unless all parachains update their collators, because if they have multiple cores assigned, an unupgraded collator could build a collation for the same core, because it's not respecting the core selector.

I still think that maybe it should now be a runtime configuration item.

Reply:
> I would not worry about parachains having multiple cores assigned, the chances that they have them is if they are using elastic scaling, which means they have read this doc and know they need all collators to be 2509+

anybody can assign cores to a parachain, so someone could do this to affect the parablock throughput

Reply:
If the parachain slot is 6s, you'd basically have old collators not building on all cores, and then an upgraded one building on all the cores. I think what you say can only happen if parachain slot time is < 6s.